src-java/kilda-pce/src/test/java/org/openkilda/pce/finder/DiversityPathFinderTest.java

```diff
@@ -0,0 +1,370 @@
+import java.util.List;
+
+@RunWith(Parameterized.class)
+public class DiversityPathFinderTest {
+    public static final SwitchId SWITCH_ID_1 = new SwitchId(1);
+    public static final SwitchId SWITCH_ID_2 = new SwitchId(2);
+    public static final SwitchId SWITCH_ID_3 = new SwitchId(3);
+    public static final SwitchId SWITCH_ID_4 = new SwitchId(4);
+    public static final SwitchId SWITCH_ID_5 = new SwitchId(5);
+    public static final Switch switchA = Switch.builder().switchId(SWITCH_ID_1).build();
+    public static final Switch switchB = Switch.builder().switchId(SWITCH_ID_2).build();
+    public static final Switch switchC = Switch.builder().switchId(SWITCH_ID_3).build();
+    public static final Switch switchD = Switch.builder().switchId(SWITCH_ID_4).build();
+    public static final Switch switchE = Switch.builder().switchId(SWITCH_ID_5).build();
+    public static final PathId FORWARD_PATH_ID = new PathId("path_id_1");
+    public static final PathId REVERSE_PATH_ID = new PathId("path_id_2");
+    public static final String DIVERSE_GROUP_ID = "group_id_1";
+
+    @Mock
+    private PathComputerConfig config;
+    @Mock
+    private RepositoryFactory repositoryFactory;
+    @Mock
+    private IslRepository islRepository;
+    @Mock
+    private FlowPathRepository flowPathRepository;
+
+    private AvailableNetworkFactory availableNetworkFactory;
+
+    @Parameter
+    public PathComputationStrategy pathComputationStrategy;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+
+        when(config.getDiversitySwitchLatency()).thenReturn(10000L);
+        when(config.getDiversityIslLatency()).thenReturn(10000L);
+        when(config.getDiversityIslCost()).thenReturn(10000);
+        when(config.getDiversitySwitchCost()).thenReturn(10000);
+        when(config.getDiversityPopIslCost()).thenReturn(10000);
+
+        when(repositoryFactory.createIslRepository()).thenReturn(islRepository);
+        when(repositoryFactory.createFlowPathRepository()).thenReturn(flowPathRepository);
+
+        availableNetworkFactory = new AvailableNetworkFactory(config, repositoryFactory);
+    }
+
+    @Test
+    public void shouldChooseSamePath() throws RecoverableException, UnroutableFlowException {
+        // Topology:
+        // A----B----C     Already created flow: A-B-C
+        //                 Requested flow: A-B-C
+
+        List<IslImmutableView> isls = new ArrayList<>();
+        isls.addAll(getBidirectionalIsls(switchA, 1, switchB, 2));
+        isls.addAll(getBidirectionalIsls(switchB, 3, switchC, 4));
+
+        List<PathSegment> forwardSegments = new ArrayList<>();
+        forwardSegments.add(PathSegment.builder().pathId(FORWARD_PATH_ID).srcSwitch(switchA).srcPort(10)
+                .destSwitch(switchB).destPort(10).build());
+        forwardSegments.add(PathSegment.builder().pathId(FORWARD_PATH_ID).srcSwitch(switchB).srcPort(10)
+                .destSwitch(switchC).destPort(10).build());
```

Comment:
switch B has 2 path segments in port 10. A-B and B-C. looks strange

Reply:
Fixed